apps/timetable/admin/ui/js/index.js

```diff
@@ -0,0 +1,411 @@
+     *
+     * @private
+     */
+    var getTripos = function() {
+        var appId = gh.data.me && gh.data.me.AppId ? gh.data.me.AppId : null;
+        gh.api.orgunitAPI.getOrgUnits(appId, false, null, ['course', 'subject', 'part'], function(err, data) {
+            if (err) {
+                return gh.api.utilAPI.notification('Fetching triposes failed.', 'An error occurred while fetching the triposes.', 'error');
+            }
+
+            triposData.courses = _.filter(data.results, function(course) {
+                return course.type === 'course';
+            });
+
+            triposData.subjects = _.filter(data.results, function(subject) {
+                return subject.type === 'subject';
+            });
+
+            triposData.parts = _.filter(data.results, function(part) {
+                return part.type === 'part';
+            });
+
+            // Sort the data before displaying it
+            triposData.courses.sort(gh.api.utilAPI.sortByDisplayName);
+            triposData.subjects.sort(gh.api.utilAPI.sortByDisplayName);
+            triposData.parts.sort(gh.api.utilAPI.sortByDisplayName);
+
+            // Set up the tripos picker after all data has been retrieved
+            setUpTriposPicker();
+        });
+    };
+
+    /**
+     * Get the user's triposes
+     *
+     * @private
+     */
+    var getUserTripos = function() {
+        /* TODO: replace this by available parts for the admin */
+        gh.api.utilAPI.renderTemplate($('#gh-main-tripos-template'), {
+            'data': null
+        }, $('#gh-main'));
+    };
+
+    /**
+     * Render the header
+     *
+     * @private
+     */
+    var renderHeader = function() {
+        gh.api.utilAPI.renderTemplate($('#gh-header-template'), {
+            'gh': gh
+        }, $('#gh-header'));
+    };
+
+    /**
+     * Render the help section
+     *
+     * @private
+     */
+    var renderHelp = function() {
+        gh.api.utilAPI.renderTemplate($('#gh-help-template'), {
+            'gh': gh
+        }, $('#gh-main'));
+    };
+
+    /**
+     * Render the login form
+     *
+     * @private
+     */
+    var renderLoginForm = function() {
+        $('#gh-subheader, #gh-content-description').height(350);
+        gh.api.utilAPI.renderTemplate($('#gh-login-template'), {
+            'gh': gh
+        }, $('#gh-subheader'));
+    };
+
+    /**
+     * Render the tripos pickers
+     *
+     * @private
+     */
+    var renderPickers = function() {
+        gh.api.utilAPI.renderTemplate($('#gh-subheader-pickers-template'), {
+            'gh': gh
+        }, $('#gh-subheader'));
+    };
+
+    /**
+     * Set up the modules of events in the sidebar
+     *
+     * @param  {Event}     ev      Standard jQuery event
+     * @param  {Object}    data    Data object describing the selected part to fetch modules for
+     * @private
+     */
+    var setUpModules = function(ev, data) {
```

Comment:
I'm not sure if this would make things unnecessarily complex but it feels like we should create individual components for the tripos and part pickers to avoid duplication.